internal/controller/csiaddons/networkfence_controller.go

```diff
@@ -325,3 +347,45 @@ func (r *NetworkFenceReconciler) getNetworkFenceClient(ctx context.Context, driv
 
 	return nil, fmt.Errorf("leading CSIAddonsNode %q for driver %q does not support NetworkFence", conn.Name, drivername)
 }
+
+func (r *NetworkFenceReconciler) getNetworkFenceInstance(
+	ctx context.Context,
+	logger logr.Logger,
+	nf *csiaddonsv1alpha1.NetworkFence) (NetworkFenceInstance, error) {
+	nfInstance := NetworkFenceInstance{
+		reconciler: r,
+		instance:   nf,
+	}
+	var err error
+
+	if nf.Spec.NetworkFenceClassName == "" {
```

Comment:
We should deprecate it, but still support the older spec for a couple of  releases. IOW, it should not be a hard error if the class is empty.

Reply:
The condition is true when classname is not specified, in that case we log a warning and use the existing values from NF only.

It is not a hard error, just a logger warning.